kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/main/java/org/kie/server/remote/rest/common/resource/KieServerRestImpl.java

```diff
@@ -75,8 +81,26 @@ public Response getInfo(@Context HttpHeaders headers) {
     @GET
     @Path("containers")
     @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
-    public Response listContainers(@Context HttpHeaders headers) { 
-        return createCorrectVariant(server.listContainers(), headers);
+    public Response listContainers(@Context HttpHeaders headers,
+                                   @QueryParam("groupId") String groupId,
+                                   @QueryParam("artifactId") String artifactId,
+                                   @QueryParam("version") String version,
+                                   @QueryParam("status") String status) {
+        ReleaseIdFilter releaseIdFilter = new ReleaseIdFilter.Builder()
+                .groupId(groupId)
+                .artifactId(artifactId)
+                .version(version)
+                .build();
+        KieContainerStatusFilter statusFilter;
+        // in case no status specified means that the filter should accept all the states
+        if (status == null) {
+            statusFilter = KieContainerStatusFilter.ACCEPT_ALL;
+        } else {
+            statusFilter = KieContainerStatusFilter.parseFromString(status);
+        }
```

Comment:
maybe instead of this if else parseFromString could return ACCEPT_ALL if the argument is null or empty, wdyt?